### PR TITLE
[14.0][MIG]purchase_requisition_auto_rfq

### PR DESCRIPTION
**Pull request by anddago78**
This module adds a button on the purchase requisition form to create RFQs
using the suppliers from the products listed in the requisition.

MT-2584 @moduon @rafaelbn @Shide @yajo  please review :)

### CODE REVIEW

**Review by fcvalgar — APPROVED** [review]
LGTM, Thank you @anddago78

File: purchase_requisition_auto_rfq/models/purchase_requisition.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        which an existing product.supplierinfo record exist for that product.
+        """
+        self.ensure_one()
+        seller_products, products_without_supplier = self._get_sellers()
+        if products_without_supplier:
+            self._post_products_without_supplier_message(products_without_supplier)
+        return self._create_rfqs(seller_products)
+
+    def _get_sellers(self):
+        seller_products = defaultdict(set)
```
Comment (Shide) [inline purchase_requisition_auto_rfq/models/purchase_requisition.py]: Can be done with 
```suggestion
        seller_products = defaultdict(self.env["product.product"])
```
and change below
```python
seller_products[seller.name.id] |= line.product_id
```

But in the end (`_remove_lines_without_official_supplier` function)  doesn't impacts too much and will be the same code.

**Review by Shide — APPROVED** [review]
LGTM. Code and Functional review.

File: purchase_requisition_auto_rfq/models/purchase_requisition.py [hunk 2]
```diff
@@ -58,7 +57,11 @@ def _post_products_without_supplier_message(self, products):
 
     def _create_purchase_order(self, seller_id):
         vals = {"partner_id": seller_id}
-        ctx = {"default_requisition_id": self.id}
+        if self.user_id:
+            uid = self.user_id.id
+        else:
+            uid = self.env.user.id
+        ctx = {"default_requisition_id": self.id, "default_user_id": uid}
```
Comment (yajo) [inline purchase_requisition_auto_rfq/models/purchase_requisition.py]: Please extract this change in a separate commit, because it is a new functionality.

**Comment by anddago78** [comment]
All comments attended

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
This one seems ready to merge. @rafaelbn do you have permissions?

**Review by Shide — APPROVED** [review]

**Review by yajo — APPROVED** [review]
Reviewed new improvements.

**Review by rafaelbn — COMMENTED** [review]
- [x] Just improve README a little bit but functionally 👍 
- [x] I think you can add Moduon as author in this case
- [x] I think you can add yourself @anddago78 as contributor AND maintainer

Thank you all!

**Comment by yajo** [comment]
Please some merge? @OCA/procurement-purchase-maintainers

**Review by yajo — APPROVED** [review]

**Comment by rafaelbn** [comment]
Are my comments attended @anddago78  ? https://github.com/OCA/purchase-workflow/pull/1817#pullrequestreview-1359869609

**Comment by anddago78** [comment]
> * [ ]  Just improve README a little bit but functionally +1
> * [ ]  I think you can add Moduon as author in this case
> * [ ]  I think you can add yourself @anddago78 as contributor AND maintainer
> 
> Thank you all!

All comments are attended!

**Comment by HviorForgeFlow** [comment]
@anddago78 you must run pre-commit locally to fix pip setup installation when module name changes

**Comment by anddago78** [comment]
Thanks @HviorForgeFlow for your comment!

**Review by HviorForgeFlow — APPROVED** [review]
